src/main/java/net/minestom/server/entity/metadata/AbstractAgeableMeta.java

```diff
@@ -5,31 +5,26 @@
 import net.minestom.server.entity.Metadata;
 import org.jetbrains.annotations.NotNull;
 
-public class AgeableMobMeta extends PathfinderMobMeta {
+public abstract class AbstractAgeableMeta extends PathfinderMobMeta implements AgeableMeta {
     public static final byte OFFSET = PathfinderMobMeta.MAX_OFFSET;
     public static final byte MAX_OFFSET = OFFSET + 1;
 
-    protected AgeableMobMeta(@NotNull Entity entity, @NotNull Metadata metadata) {
+    protected AbstractAgeableMeta(@NotNull Entity entity, @NotNull Metadata metadata) {
         super(entity, metadata);
     }
 
+    @Override
     public boolean isBaby() {
         return super.metadata.getIndex(OFFSET, false);
     }
 
+    @Override
     public void setBaby(boolean value) {
-        if (isBaby() == value) {
-            return;
-        }
+        if (isBaby() == value) return;
         this.consumeEntity((entity) -> {
-            BoundingBox bb = entity.getBoundingBox();
-            if (value) {
-                double width = bb.width() / 2;
-                entity.setBoundingBox(width, bb.height() / 2, width);
-            } else {
-                double width = bb.width() * 2;
-                entity.setBoundingBox(width, bb.height() * 2, width);
-            }
+            BoundingBox bb = entity.getEntityType().registry().boundingBox();
+            if (value) entity.setBoundingBox(bb.width() * 0.5, bb.height() * 0.5, bb.depth() * 0.5);
```

Comment:
I think this could be better done with something along the lines of `protected abstract double boundingBoxMultiplier();`, and then certain metas override that instead of repeating this chunk of code.

Reply:
I'm not a fan of this. Not all entities extend AbstractAgeableMeta for a reason unfortunately (thanks mojang). Making metadata extend this arbitrary class would add api confusion imo. Could be delegated to the EntityUtils class but I think the current refactor is already a good start